pandas/core/apply.py

```diff
@@ -111,8 +111,11 @@ def get_result(self):
 
         # string dispatch
         if isinstance(self.f, compat.string_types):
-            self.kwds['axis'] = self.axis
-            return getattr(self.obj, self.f)(*self.args, **self.kwds)
+            func = getattr(self.obj, self.f)
```

Comment:
much better, and now IIRC we actually do something like this in groupby apply (for the same reason). can you add some comments here on what is going on